OpenRA.Mods.Common/Traits/Buildings/Refinery.cs

```diff
@@ -105,6 +114,10 @@ public void GiveResource(int amount)
 			else
 				amount = playerResources.ChangeCash(amount);
 
+			var externalMultipliers = self.World.ActorsWithTrait<ExternalResourceMultiplier>().Where(x => x.Actor.Owner == self.Owner).Select(x => x.Trait);
```

Comment:
As discussed in IRC it would be best if this was included in a RA2-specific `Refinery` subclass, until such a time that the refinery logic can be properly rewritten.  This PR can make the appropriate methods virtual.